transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemAdd.java

```diff
@@ -446,16 +447,31 @@ private void performRecoveryEnvBoottime(OperationContext context, ModelNode mode
                     .install();
         }
 
+        final String nodeIdentifier = TransactionSubsystemRootResourceDefinition.NODE_IDENTIFIER.resolveModelAttribute(context, model).asString();
+        // install JTA environment bean service
+        final JTAEnvironmentBeanService jtaEnvironmentBeanService = new JTAEnvironmentBeanService(nodeIdentifier);
+        context.getServiceTarget().addService(TxnServices.JBOSS_TXN_JTA_ENVIRONMENT, jtaEnvironmentBeanService)
+                .setInitialMode(Mode.ACTIVE)
+                .install();
+
         final XATerminatorService xaTerminatorService;
         final ExtendedJBossXATerminatorService extendedJBossXATerminatorService;
 
+        JTAEnvironmentBeanService jtaBeanService =
```

Comment:
What is the purpose of this? You just installed this service a few lines above, what you get from this lookup should be the same as jtaEnvironmentBeanService

Reply:
only my omission. Thank you for spotting it. I was experimenting with retrieving service on a different place and I made a dummy copy&paste error here.